core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -188,39 +190,30 @@ oppia.factory('oppiaPlayerService', [
         playerTranscriptService.init();
 
         if (_editorPreviewMode) {
-          var explorationDataUrl = UrlInterpolationService.interpolateUrl(
-            '/createhandler/data/<exploration_id>', {
-              exploration_id: _explorationId
-            });
-          $http.get(explorationDataUrl, {
-            params: {
-              apply_draft: true
-            }
-          }).then(function(response) {
-            exploration = ExplorationObjectFactory.create(response.data);
+          EditableExplorationBackendApiService.fetchExploration(
+            _explorationId, true).then(function(data) {
+            exploration = ExplorationObjectFactory.create(data);
             exploration.setInitialStateName(initStateName);
             initParams(manualParamChanges);
             _loadInitialState(successCallback);
           });
         } else {
-          var explorationDataUrl = UrlInterpolationService.interpolateUrl(
-            '/explorehandler/init/<exploration_id>', {
-              exploration_id: _explorationId
-            }) + (version ? '?v=' + version : '');
-          $http.get(explorationDataUrl).then(function(response) {
-            var data = response.data;
-            exploration = ExplorationObjectFactory.create(data.exploration);
-            version = data.version;
+          ReadOnlyExplorationBackendApiService.loadExploration(
+            _explorationId, version ? version : null).then(
```

Comment:
I am wrestling with this. Do I leave this as loadExploration? Or do I add a loadLatestExploration check. I'll go with the latter and see if you (@BenHenning) like the way it looks.